feat: Add job name/id to job waitoutput and improve timestamp output for 'logs' command

prestomation opened the pull request:
I received some offline feedback for the previous change adding these commands.

I have added job name/id to the output of the 'wait' command, as if you use the default job there is no indication of which job you are waiting on.

I've improved the timestamp handling in the logs command, it now returns ISO8601 and there is a '--timezone local' switch which will print out in local timezone:

Fixes: *<insert link to GitHub issue here>*

### What was the problem/requirement? (What/Why)
`deadline job wait` had no output indicting which job was being waited on
`deadline job logs` output no precision timestamps with ambiguous timezones


### What was the solution? (How)
`deadline job wait` now prints the job id and name
`deadline job logs` now prints iso8601 timestamps, and has a `--timezone local` switch to use the system timezone.


### What is the impact of this change?
More precise information for users of these commands
 

### How was this change tested?

See [DEVELOPMENT.md](https://github.com/aws-deadline/deadline-cloud/blob/mainline/DEVELOPMENT.md#testing) for information on running tests.

- Have you run the unit tests? yes
- Have you run the integration tests? no
- Have you made changes to the `download` or `asset_sync` modules? If so, then it is highly recommended
  that you ensure that the docker-based unit tests pass. no

### Was this change documented?

- Are relevant docstrings in the code base updated? yes
- Has the README.md been updated? If you modified CLI arguments, for instance. yes

### Does this PR introduce new dependencies?

This library is designed to be integrated into third-party applications that have bespoke and customized deployment environments. Adding dependencies will increase the chance of library version conflicts and incompatabilities. Please evaluate the addition of new dependencies. See the [Dependencies](https://github.com/aws-deadline/deadline-cloud/blob/mainline/DEVELOPMENT.md#dependencies) section of DEVELOPMENT.md for more details.

*   [ ] This PR adds one or more new dependency Python packages. I acknowledge I have reviewed the considerations for adding dependencies in [DEVELOPMENT.md](https://github.com/aws-deadline/deadline-cloud/blob/mainline/DEVELOPMENT.md#dependencies).
*   [x ] This PR does not add any new dependencies.

### Is this a breaking change?

A breaking change is one that modifies a public contract in a way that is not backwards compatible. See the 
[Public Contracts](https://github.com/aws-deadline/deadline-cloud/blob/mainline/DEVELOPMENT.md#public-contracts) section
of the DEVELOPMENT.md for more information on the public contracts.

If so, then please describe the changes that users of this package must make to update their scripts, or Python applications.


Technically I guess? but the base change was merged just a few hours ago

### Does this change impact security?

No

- Does the change need to be threat modeled? For example, does it create or modify files/directories that must only be readable by the process owner?
    - If so, then please label this pull request with the "security" label. We'll work with you to analyze the threats.

----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

leongdl commented:
DCO is failing, please use `git commit --amend -s`

leongdl commented:
```
FAILED test/unit/deadline_client/cli/test_cli_job_logs.py::test_cli_job_logs_json_error - assert 'error' in 'The AWS Deadline Cloud CLI encountered the following exception:\nTraceback (most recent call last):\n  File "/Users/r... _endpoint_for_partition\n    raise NoRegionError()\nbotocore.exceptions.NoRegionError: You must specify a region.\n\n'
 +  where 'The AWS Deadline Cloud CLI encountered the following exception:\nTraceback (most recent call last):\n  File "/Users/r... _endpoint_for_partition\n    raise NoRegionError()\nbotocore.exceptions.NoRegionError: You must specify a region.\n\n' = <Result SystemExit(1)>.output
```

Does this mean it needs to be integ tests?

evanspearman-a commented:
I don't think this should be considered a breaking change. Adding a field to json _output_ shouldn't be considered breaking in this sort of context.

prestomation commented:
I've fixed up the tests. it's not an integ test but the way the SDK was being used it tried to load creds even though they aren't used. I've added an autouse fixture to override with an empty aws/config file so this doesn't pollute tests in the future. 

The breaking change is not due to adding the new field, but because the timestamp format is changing which someone could be relying on.

prestomation commented:
The SonarQube failure is about some test duplication which I'm not inclined to refactor but happy to do so if there are existing patterns I should follow

leongdl commented:
> I've fixed up the tests. it's not an integ test but the way the SDK was being used it tried to load creds even though they aren't used. I've added an autouse fixture to override with an empty aws/config file so this doesn't pollute tests in the future.
> 
> The breaking change is not due to adding the new field, but because the timestamp format is changing which someone could be relying on.

If this is a breaking changes, please use `feat!` in the commit message for our version bump workflows.

prestomation commented:
Hm actually I will leave it as it is because the change it 'breaks' has not yet been released.
